File: Math/263_UglyNumber.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-
# @Author: xuezaigds@gmail.com


class Solution(object):
    def isUgly(self, num):
        for p in [2, 3, 5]:
            while num and num % p == 0:
                num /= p
        return num == 1

"""
-2147483648
1
0
14
8
"""

File: Math/263_UglyNumber.cpp
/*
 * @Author: xuezaigds@gmail.com
 * @Last Modified time: 2016-04-15 22:23:06
 */

class Solution {
public:
    bool isUgly(int num) {
        int prime[] = {2,3,5};
        for(auto &n : prime){
            while(num!=0 && num%n==0)   num/=n;
        }
        return num==1;
    }
};

/*
-2147483648
1
0
14
8
*/
